tests/network/port_forward.go

```diff
@@ -0,0 +1,145 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = SIGDescribe("[Serial] Port-forward", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("VMI With masquerade binding", func() {
+		const (
+			localPort       = 1500
+			declaredPort    = 1501
+			nonDeclaredPort = 1502
+		)
+
+		var portForwardCmd *exec.Cmd
+
+		When("performing port-forward from a local port to a VMI's pod declared port", func() {
+			var (
+				ports = []v1.Port{
+					{Port: declaredPort},
+				}
+				vmiWithDeclaredPort *v1.VirtualMachineInstance
+			)
+
+			BeforeEach(func() {
+				vmiWithDeclaredPort = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, ports)
+				tests.StartHTTPServer(vmiWithDeclaredPort, declaredPort)
+
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmiWithDeclaredPort, tests.NamespaceTestDefault)
+				Expect(vmiPod).ToNot(BeNil())
+				portForwardCmd = portForwardCommand(vmiPod.Name, vmiPod.Namespace, localPort, declaredPort)
+				Expect(portForwardCmd.Start()).To(Succeed())
+			})
+
+			AfterEach(func() {
+				Expect(portForwardCmd.Process.Kill()).To(Succeed())
+			})
+
+			It("should reach the vmi", func() {
+				vmiWithDeclaredPort, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmiWithDeclaredPort.Name, &metav1.GetOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", declaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 20*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+
+		When("performing port-forward from a local port to a VMI's pod non-declared port", func() {
+			var vmiWithoutDeclaredPort *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmiWithoutDeclaredPort = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, []v1.Port{})
+				tests.StartHTTPServer(vmiWithoutDeclaredPort, nonDeclaredPort)
+
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmiWithoutDeclaredPort, tests.NamespaceTestDefault)
+				Expect(vmiPod).ToNot(BeNil())
+				portForwardCmd = portForwardCommand(vmiPod.Name, vmiPod.Namespace, localPort, nonDeclaredPort)
+				Expect(portForwardCmd.Start()).To(Succeed())
+			})
+
+			AfterEach(func() {
```

Comment:
You could move this to the upper Context, and then you don't need it twice

Reply:
Tnx. Done.